_rules/element-lang-valid-de46e4.md

```diff
@@ -19,15 +19,18 @@ authors:
 
 ## Applicability
 
-This rules applies to any HTML element that is a [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) of a `body` element and has a `lang` attribute that is neither empty ("") nor only [ASCII whitespace](https://infra.spec.whatwg.org/#ascii-whitespace).
+This rules applies to any HTML element that:
+
+- has a [node document](https://dom.spec.whatwg.org/#concept-node-document) with a [content type](https://dom.spec.whatwg.org/#concept-document-content-type) of `text/html`; and
```

Comment:
I think it should use [type](https://dom.spec.whatwg.org/#concept-document-type) instead of [content-type](https://dom.spec.whatwg.org/#concept-document-content-type) since type is what affects the APIs. 

This covers documents nodes without a content-type created by JavaScript APIs like createDocumentFragment or createDocument

Reply:
Documents will always have a content type, even when created via JavaScript. For [`createDocument()`](https://dom.spec.whatwg.org/#dom-domimplementation-createdocument), the content type will either be `application/xhtml+xml`, `image/svg+xml`, or `application/xml`, whereas for [`createHTMLDocument()`](https://dom.spec.whatwg.org/#dom-domimplementation-createhtmldocument) the content type will always be `text/html`. For [`createDocumentFragment()`], its node document will be the context object on which the API was invoked so the created `DocumentFragment` will effectively inherit the content type of its content object.